tests/ui/macros/issue-102878.stderr

```diff
@@ -2,7 +2,7 @@ error: mismatched closing delimiter: `)`
   --> $DIR/issue-102878.rs:1:35
    |
 LL | macro_rules!test{($l:expr,$_:r)=>({const:y y)}
-   |                                  -^         ^ mismatched closing delimiter
+   |                                  -^         ^ mismatched closing delimiter, may missing open `(`
    |                                  ||
```

Comment:
this is worse too, we are not sure should we should replace the left `{` to `(` or replace the right `)` to `}`.

so, the better way is do not suggest any direction.

I think we should only suggest missing a open(or closing) delimiter in the scenario that if we added a open(or closing) delimiter, there will be no error.